site/app/controllers/admin/PlagiarismController.php

```diff
@@ -68,9 +68,35 @@ private function getGradeablesFromPriorTerm() {
     }
 
 
-    private function getRankings($gradeable_id) {
+    private function getOverallRankings($gradeable_id) {
         $course_path = $this->core->getConfig()->getCoursePath();
-        $file_path = $course_path . "/lichen/ranking/" . $gradeable_id . ".txt";
+        $file_path = $course_path . "/lichen/ranking/" . $gradeable_id . "/overall_ranking.txt";
+        if (!file_exists($file_path)) {
+            $this->core->addErrorMessage("Plagiarism Detection job is running for this gradeable.");
+            $this->core->redirect($this->core->buildCourseUrl(['plagiarism']));
+        }
+        $test = file_get_contents($file_path);
+        if (file_get_contents($file_path) == "") {
+            $this->core->addSuccessMessage("There are no matches(plagiarism) for the gradeable with current configuration");
+            //$this->core->redirect($this->core->buildCourseUrl(['plagiarism']));
+        }
+        $content = file_get_contents($file_path);
+        $content = trim(str_replace(["\r", "\n"], ' ', $content));
+        $rankings = preg_split('/ +/', $content);
+        $rankings = array_chunk($rankings, 3);
+        return $rankings;
+    }
+
+    /**
+     * Returns a ranking of users by percent match with user 1 (used for determining the rightmost dropdown list)
+     * @param string $gradeable_id
+     * @param string $user_id_1
+     * @param string $user_1_version
+     * @return array
+     */
+    private function getRankingsForUser($gradeable_id, $user_id_1, $user_1_version) {
```

Comment:
Since this function already has the parameters annotated it would be good to add type hints to this and other functions added to make PHP enforce types better and improve static analysis

https://submitty.org/developer/coding_style_guide/php#type-declarations